### ISSUE TITLE
GlobalISel: infinite loop trying to legalize G_BUILD_VECTOR

### ISSUE THREAD

**Issue by jayfoad** [llvm/llvm-project#79578]
```
; RUN: llc -global-isel=1 -mtriple=amdgcn -mcpu=gfx1030 < %s
define void @main(<19 x i32> %arg) {
bb:
  %i = bitcast <19 x i32> %arg to <38 x i16>
  %i1 = extractelement <38 x i16> %i, i64 0
  %i2 = icmp eq i16 %i1, 0
  %i3 = zext i1 %i2 to i32
  %i4 = bitcast i32 %i3 to float
  %i5 = insertelement <4 x float> zeroinitializer, float %i4, i64 0
  call void @llvm.amdgcn.image.store.2d.v4f32.i32(<4 x float> %i5, i32 0, i32 0, i32 0, <8 x i32> zeroinitializer, i32 0, i32 0)
  ret void
}
declare void @llvm.amdgcn.image.store.2d.v4f32.i32(<4 x float>, i32 immarg, i32, i32, <8 x i32>, i32 immarg, i32 immarg)
```

**Comment by petar-avramovic** [comment]
The problem here is that 
  `%0:_(<19 x s32>) = G_BUILD_VECTOR` is not legal for `<19 x s32>`
while the user of `%0:_(<19 x s32>)` is legal for `<19 x s32>`. User is `G_BITCAST`
```
  %20:_(<38 x s16>) = G_BITCAST %0:_(<19 x s32>)
  %32:_(<19 x s32>) = G_BITCAST %20:_(<38 x s16>)
```
`G_BUILD_VECTOR` waits for user to get legalized, legalization process should create `G_UNMERGE_VALUES (<19 x s32>)` and this two should be combined away.
Note: artifact combiner is not allowed to look through `G_BITCAST` so `G_BITCAST` has to be legalized by the target.
The now allowed to look through comes from how `G_BITCAST` was described in globalisel documentation if I remember correctly.

In summary `G_BITCAST` should not be legal for `<19 x s32>` and `<38 x s16>`

Good first step is to first properly list all all legal types.
There is a PR that tried to implement this
https://github.com/llvm/llvm-project/pull/68189
